pkgs/build-support/bintools-wrapper/ld-wrapper.sh

```diff
@@ -134,7 +145,7 @@ then
     done
 fi
 
-if [ -e "@out@/nix-support/dynamic-linker-m32" ] && (( "$link32" )); then
+if [[ "$link32" = "0" && "$setDynamicLinker" = 1 && -e "@out@/nix-support/dynamic-linker-m32" ]]; then
```

Comment:
As pointed out in #107253, `(( "$link32" ))` is an arithmetic expansion that returns true (status 0) if `$link32` is nonzero, not if it’s zero.